dav1d-sys/src/lib.rs

```diff
@@ -3,7 +3,664 @@
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Dav1dDataProps {
+    pub timestamp: i64,
+    pub duration: i64,
+    pub offset: i64,
+    pub size: usize,
+    pub user_data: Dav1dUserData,
+}
+
+pub const Dav1dObuType_DAV1D_OBU_SEQ_HDR: Dav1dObuType = 1;
+pub const Dav1dObuType_DAV1D_OBU_TD: Dav1dObuType = 2;
+pub const Dav1dObuType_DAV1D_OBU_FRAME_HDR: Dav1dObuType = 3;
+pub const Dav1dObuType_DAV1D_OBU_TILE_GRP: Dav1dObuType = 4;
+pub const Dav1dObuType_DAV1D_OBU_METADATA: Dav1dObuType = 5;
+pub const Dav1dObuType_DAV1D_OBU_FRAME: Dav1dObuType = 6;
+pub const Dav1dObuType_DAV1D_OBU_REDUNDANT_FRAME_HDR: Dav1dObuType = 7;
+pub const Dav1dObuType_DAV1D_OBU_PADDING: Dav1dObuType = 15;
+
+pub type Dav1dObuType = ::std::os::raw::c_uint;
+pub const Dav1dTxfmMode_DAV1D_TX_4X4_ONLY: Dav1dTxfmMode = 0;
+pub const Dav1dTxfmMode_DAV1D_TX_LARGEST: Dav1dTxfmMode = 1;
+pub const Dav1dTxfmMode_DAV1D_TX_SWITCHABLE: Dav1dTxfmMode = 2;
+pub const Dav1dTxfmMode_DAV1D_N_TX_MODES: Dav1dTxfmMode = 3;
+
+pub type Dav1dTxfmMode = ::std::os::raw::c_uint;
+pub const Dav1dFilterMode_DAV1D_FILTER_8TAP_REGULAR: Dav1dFilterMode = 0;
+pub const Dav1dFilterMode_DAV1D_FILTER_8TAP_SMOOTH: Dav1dFilterMode = 1;
+pub const Dav1dFilterMode_DAV1D_FILTER_8TAP_SHARP: Dav1dFilterMode = 2;
+pub const Dav1dFilterMode_DAV1D_N_SWITCHABLE_FILTERS: Dav1dFilterMode = 3;
+pub const Dav1dFilterMode_DAV1D_FILTER_BILINEAR: Dav1dFilterMode = 3;
+pub const Dav1dFilterMode_DAV1D_N_FILTERS: Dav1dFilterMode = 4;
+pub const Dav1dFilterMode_DAV1D_FILTER_SWITCHABLE: Dav1dFilterMode = 4;
+
+pub type Dav1dFilterMode = ::std::os::raw::c_uint;
+pub const Dav1dAdaptiveBoolean_DAV1D_OFF: Dav1dAdaptiveBoolean = 0;
+pub const Dav1dAdaptiveBoolean_DAV1D_ON: Dav1dAdaptiveBoolean = 1;
+pub const Dav1dAdaptiveBoolean_DAV1D_ADAPTIVE: Dav1dAdaptiveBoolean = 2;
+
+pub type Dav1dAdaptiveBoolean = ::std::os::raw::c_uint;
+pub const Dav1dRestorationType_DAV1D_RESTORATION_NONE: Dav1dRestorationType = 0;
+pub const Dav1dRestorationType_DAV1D_RESTORATION_SWITCHABLE: Dav1dRestorationType = 1;
+pub const Dav1dRestorationType_DAV1D_RESTORATION_WIENER: Dav1dRestorationType = 2;
+pub const Dav1dRestorationType_DAV1D_RESTORATION_SGRPROJ: Dav1dRestorationType = 3;
+
+pub type Dav1dRestorationType = ::std::os::raw::c_uint;
+pub const Dav1dWarpedMotionType_DAV1D_WM_TYPE_IDENTITY: Dav1dWarpedMotionType = 0;
+pub const Dav1dWarpedMotionType_DAV1D_WM_TYPE_TRANSLATION: Dav1dWarpedMotionType = 1;
+pub const Dav1dWarpedMotionType_DAV1D_WM_TYPE_ROT_ZOOM: Dav1dWarpedMotionType = 2;
+pub const Dav1dWarpedMotionType_DAV1D_WM_TYPE_AFFINE: Dav1dWarpedMotionType = 3;
+
+pub type Dav1dWarpedMotionType = ::std::os::raw::c_uint;
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct Dav1dWarpedMotionParams {
+    pub type_: Dav1dWarpedMotionType,
+    pub matrix: [i32; 6usize],
+    pub u: Dav1dWarpedMotionParams__bindgen_ty_1,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub union Dav1dWarpedMotionParams__bindgen_ty_1 {
```

Comment:
This is all still based on the bindgen struct definitions and I would clean this up a bit more with regards to the naming